Services/Export/classes/class.ilImport.php

```diff
@@ -292,6 +292,15 @@ protected function doImportObject($dir, $a_type, $a_component = "", $a_tmpdir =
 			$comp = $expfile["component"];
 			$class = ilImportExportFactory::getImporterClass($comp);
 
+			// log a warning for inactive page component plugins, but continue import
```

Comment:
This still bothers me. The comment adresses a slot specific thing "page component slot" in a general class (ilImport). Would the same not work for other slots that support import/export capabilities?

Reply:
Hi Alex, 

I added some comments in ilImportExportFactory::getImporterClass() to make things more obvious. The concept would be: 

a) if this function throwns an exception, then the import should fail. 
b) if the return is empty then the importer should continue the import without the component. 

Case b) currently only happens for page component plugins that export their own data and that are not active in the target platform. All other plugins will throw the exception. 

This might be imroved in the future by a more sophisticated exception handling (e.g. let the user decide if a partial import should be done).  But as a short fix for the release the concept should work.